sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -372,11 +367,28 @@ object PartitioningUtils {
       suspiciousPaths.map("\t" + _).mkString("\n", "\n", "")
   }
 
+  // scalastyle:off line.size.limit
   /**
-   * Converts a string to a [[Literal]] with automatic type inference.  Currently only supports
-   * [[IntegerType]], [[LongType]], [[DoubleType]], [[DecimalType]], [[DateType]]
+   * Converts a string to a [[Literal]] with automatic type inference. Currently only supports
+   * [[NullType]], [[IntegerType]], [[LongType]], [[DoubleType]], [[DecimalType]], [[DateType]]
    * [[TimestampType]], and [[StringType]].
+   *
+   * When resolving conflicts, it follows the table below:
+   *
+   * +-------------------+-------------------+-------------------+-------------------+-------------------+------------+---------------+---------------+------------+
+   * | InputA \ InputB   | NullType          | IntegerType       | LongType          | DecimalType(38,0) | DoubleType | DateType      | TimestampType | StringType |
+   * +-------------------+-------------------+-------------------+-------------------+-------------------+------------+---------------+---------------+------------+
+   * | NullType          | NullType          | IntegerType       | LongType          | DecimalType(38,0) | DoubleType | DateType      | TimestampType | StringType |
+   * | IntegerType       | IntegerType       | IntegerType       | LongType          | DecimalType(38,0) | DoubleType | StringType    | StringType    | StringType |
+   * | LongType          | LongType          | LongType          | LongType          | DecimalType(38,0) | StringType | StringType    | StringType    | StringType |
+   * | DecimalType(38,0) | DecimalType(38,0) | DecimalType(38,0) | DecimalType(38,0) | DecimalType(38,0) | StringType | StringType    | StringType    | StringType |
```

Comment:
might be good to explain why we can only see `DecimaType(38, 0)`

Reply:
Sure.